velox/type/Filter.cpp

```diff
@@ -847,4 +793,330 @@ std::unique_ptr<Filter> BigintMultiRange::mergeWith(const Filter* other) const {
       VELOX_UNREACHABLE();
   }
 }
+
+namespace {
+bool mergeUpperExclusive(
```

Comment:
mergeUpperExclusive and mergeLowerExclusive are very similar. I wonder if these can be combined into a single function:

```
// compareResult = left < right for upper, right < left for lower
bool mergeExclusive(int compareResult, bool left, bool right) {
  return compareResult == 0 ? (left || right)
                            : (compareResult < 0 ? left : right);
}
```